pkg/rest/services/cluster.go

```diff
@@ -18,12 +23,87 @@ func NewClusterService(store storeadapter.ClusterStore) *ClusterService {
 	}
 }
 
+func (s *ClusterService) GetClusterNames(c echo.Context) error {
+	clusters, err := s.store.ListClusters()
+	if err != nil {
+		return err
+	}
+	names := []string{}
+	for _, cluster := range clusters {
+		names = append(names, cluster.Name)
+	}
+	return util.AssembleResponse(c, names, nil)
+}
+
 func (s *ClusterService) GetClusters(c echo.Context) error {
 	clusters, err := s.store.ListClusters()
 	if err != nil {
 		return err
 	}
-	return c.JSON(http.StatusOK, model.ClusterListResponse{
-		Clusters: clusters,
-	})
+	return util.AssembleResponse(c, model.ClusterListResponse{Clusters: clusters}, nil)
+}
+
+func (s *ClusterService) AddCluster(c echo.Context) error {
+	clusterReq := new(apis.ClusterRequest)
+	if err := c.Bind(clusterReq); err != nil {
+		return err
+	}
+	exist, err := s.checkClusterExist(clusterReq.Name)
+	if err != nil {
+		return err
+	}
+	if exist {
+		return util.AssembleResponse(c, nil,
```

Comment:
No need for AssembleResponse() func.
Just return:

```
return c.JSON(http.StatusInternalServerError, fmt.Sprintf("cluster %s exist", clusterReq.Name))
```

Reply:
done